src/receivers/kafka/receiver.rs

```diff
@@ -203,64 +183,93 @@ impl KafkaReceiver {
+                    match decoded_result {
+                        Ok(decode_result) => {
+                            match decode_result {
+                                Ok(decoded) => {
+                                    match decoded {
+                                    // N.B - Explicitly disabling sending any metadata for now on this next commit.
+                                    // We are doing this because we are wiring in Message<T> handing with acknowledgement
+                                    // end-to-end all the way to the exporters. However, as we're not doing anything
+                                    // with the acknowledgement, we disable their creation for now out of an abundance of caution.
+                                    // This will allow us to land these changes and others while iterating on the additional pieces
+                                    // of Kafka offset tracking. Finally, once everything is in place, we can "wire up" sending the metadata
+                                    // and verify with some aggressive end-to-end tests that everything is working as expected.
+                                    // metadata: Some(payload::MessageMetadata::Kafka(kafka_metadata)),
+                                        DecodedResult::Traces { resources, metadata: _ } => {
+                                            if let Some(ref output) = self.traces_output {
+                                                let message = payload::Message::new(None, resources);
+                                                if let Err(e) = output.send(message).await {
+                                                    warn!("Failed to send traces to pipeline: {}", e);
+                                                }
+                                            }
+                                        }
+                                        DecodedResult::Metrics { resources, metadata: _ } => {
+                                            if let Some(ref output) = self.metrics_output {
+                                                let message = payload::Message::new(None, resources);
+                                                if let Err(e) = output.send(message).await {
+                                                    warn!("Failed to send metrics to pipeline: {}", e);
+                                                }
+                                            }
+                                        }
+                                        DecodedResult::Logs { resources, metadata: _ } => {
+                                            if let Some(ref output) = self.logs_output {
+                                                let message = payload::Message::new(None, resources);
+                                                if let Err(e) = output.send(message).await {
```

Comment:
So I notice we wait on the send here, which if it looks like we were also doing in the previous version, but in process_payload(). At some point we may want to move the send future back to the top select loop so that other branches, like cancellation, can occur if the send were to block.

Reply:
oh, good catch. I'll have a look in the morning.